test/runtests.jl

```diff
@@ -194,12 +194,25 @@ module Interfaces
     dive2(::Penguin, ::Vector) = 2              # Vector >: Vector{<:Integer}
     dive31(::Penguin, ::Number) = 31            # no kw argument
     dive32(::Penguin, ::Number; kw::Complex) = 32 # kw argument type is ignored!
-    dive33(::Penguin, ::Number; kw...) = 33     # any number of kw arguments
+    dive33(::Penguin, ::Int; kw...) = 33        # any number of kw arguments
     dive34(::Penguin, ::Float64) = 34           # keyword argument missing
     dive4(::Penguin, ::Integer) = 4             # Integer >: Base.Bottom
     dive5(::Penguin, ::Int) = 5                 # Int >: Bottom
     dive6(::Penguin, ::Int) = 6                 # not Int >: Number
 
+    # no contract requirements (code coverage)
+    struct Kiwi end
+
+    # weird argument types in contract specs 
+    @trait Creep
+    @implement CanCreep by creep1(a::Int=5) # argument assignment
+    @implement CanCreep by creep2({}) # invalid argument
```

Comment:
Should probably raise an error instead of considering it as a `Bottom`? 

```
julia> @macroexpand(@implement CanCreep by creep({}))
quote
    function creep end
    BinaryTraits.register(CanCreep, creep, (Base.Bottom,), (), Any)
end
```

Reply:
Ok, can do. Then I will accept only the following syntax:
```
<name> ::= <Symbol>
<argpart> ::= <name> OR <name> :: <type> OR :: <type>
<arg> ::= <argpart> OR <argpart> = <value>
<value> ::= <any Expr or Symbol>
```
and otherwise throw a Syntax exception.

Reply:
As this PR is already closed, I will open a new one for these changes.